contributed_definitions/nyaml/NXsingle_crystal.yaml

```diff
@@ -0,0 +1,31 @@
+category: base
+doc: |
+  Description of a single crystal material or a single crystalline phase in a material.
+type: group
+NXsingle_crystal(NXobject):

```

Comment:
Sorry for being pedantic here this is a too strong assumption the sample orientation is only the same orientation if the sample is a single crystal, therefore the idea is an NXms_feature can have an instance of NXrotation_set which has all references to coordinate systems, rotation, orientation, disorientation, etc.